rust/src/mask/config/mod.rs

```diff
@@ -168,9 +168,24 @@ pub struct MaskConfig {
 
 impl MaskConfig {
     /// Returns the number of bytes needed for an element of a mask object.
+    ///
+    /// # Panics
+    /// Panics if the bytes per number can't be represented as usize.
     pub(crate) fn bytes_per_number(&self) -> usize {
         let max_number = self.order() - BigUint::from(1_u8);
-        (max_number.bits() + 7) / 8
+        let bpn = (max_number.bits() + 7) / 8;
+        match mem::size_of::<usize>() {
```

Comment:
Looks like there's a cfg flag for this: https://stackoverflow.com/a/41896499/1836144
It would make the intent a bit clearer and have the benefit of happening at compile time.

Reply:
that's useful, i'll update it.

Reply:
but this just helps us with the `size_of::<usize>()` which is only part of the condition, since it is dependent on the value of `bpn` as well which in turn is not known at compile time. not sure if we can split the `match guard` then, wdyt?

Reply:
Something like this maybe:

```rust
#[cfg(target_pointer_width = "64")]
const MAX_BPN: u64 = u64::MAX as u64;

#[cfg(target_pointer_width = "32")]
const MAX_BPN: u32 = u32::MAX as u32;

#[cfg(target_pointer_width = "16")]
const MAX_BPN: u16 = u16::MAX as u16;
```

Then the condition simply becomes:
```rust
if bpn > MAX_BPN {
    panic!("not good");
}
```

Reply:
i've updated this in a similar fashion as suggested.